cdap-data-fabric/src/main/java/co/cask/cdap/store/AbstractOwnerStore.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.store;
+
+import co.cask.cdap.common.kerberos.OwnerStore;
+import co.cask.cdap.common.kerberos.SecurityUtil;
+import co.cask.cdap.proto.element.EntityType;
+import co.cask.cdap.proto.id.KerberosPrincipalId;
+import co.cask.cdap.proto.id.NamespacedEntityId;
+import com.google.common.collect.Sets;
+
+import java.util.Set;
+
+/**
+ * Abstract implementation of {@link OwnerStore} which has some shared helper function for input validation
+ */
+abstract class AbstractOwnerStore implements OwnerStore {
+
+  private static final Set<EntityType> SUPPORTED_ENTITY_TYPES = Sets.newHashSet(EntityType.NAMESPACE,
+                                                                                EntityType.APPLICATION,
+                                                                                EntityType.DATASET, EntityType.STREAM);
+
+
+  void validate(NamespacedEntityId entityId, KerberosPrincipalId principalId) {
```

Comment:
javadoc. Also, should this be `final` method? Do you expect class to override it?